### PR TITLE
[testlib] Add generic Paraview check for the cscs-supported apps into hpctestlib/apps

### PR DESCRIPTION
**Pull request by hurricane642**
This is a continuation of the work on the test library. This PR work has been done with Paraview tests. A new file base_check.py has been added, introducing the main class for the Paraview test, as well as the CSCS tests inherited from it.

### CODE REVIEW

**Comment by teojgo** [comment]
Ok to test

**Review by teojgo — CHANGES_REQUESTED** [review]
Just remove the unused imports.

File: cscs-checks/apps/paraview/paraview_check.py [hunk 1]
```diff
@@ -5,36 +5,26 @@
 
 import reframe as rfm
 import reframe.utility.sanity as sn
+from hpctestlib.apps.paraview.base_check import ParaView_BaseCheck
```
Comment (teojgo) [inline cscs-checks/apps/paraview/paraview_check.py:7]: Add also a blank line before this.

**Review by teojgo — CHANGES_REQUESTED** [review]

File: hpctestlib/apps/paraview/base_check.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    #: :default: : 'None'
+    gpu_vendor = variable(str, value='None')
+
+    #: Name of processor.
+    #:
+    #: :default: : 'None'
+    gpu_renderer = variable(str, value='None')
+
+    @run_before('sanity')
+    def set_sanity(self):
```
Comment (teojgo) [inline hpctestlib/apps/paraview/base_check.py:64]: Can you change and use the `@sanify_function` here?

**Review by teojgo — CHANGES_REQUESTED** [review]

**Review by teojgo — APPROVED** [review]

**Comment by vkarak** [comment]
This still needs substantial work, so I'm closing the PR due to time constraints. It will be added to the library at a later time.

**Pull request closed without merging**